packages/zoe/src/contracts/otcDeskReputation.js

```diff
@@ -0,0 +1,142 @@
+
+  let numTradesAttempted = 0;
+  let numTradesDefected = 0;
+
+  const makeQuote = (price, assets, timerAuthority, deadline) => {
+    let quoteActive = true;
+    E(timerAuthority).setWakeup(
+      deadline,
+      harden({ wake: () => (quoteActive = false) }),
+    );
+    const acceptQuote = seat => {
+      assert(quoteActive, details`The quote was no longer active`);
+      const { want: userWantActual, give: userGiveActual } = seat.getProposal();
+      numTradesAttempted += 1;
+      const hasInventory = Object.entries(assets).every(([keyword, amount]) => {
+        const allocatedAmount = marketMakerSeat.getAmountAllocated(
+          keyword,
+          amount.brand,
+        );
+        return zcf.getAmountMath(amount.brand).isGTE(allocatedAmount, amount);
+      });
+      if (!hasInventory) {
+        numTradesDefected += 1;
+        throw Error(
+          `The market maker did not have the inventory they promised. Their rating has been adjusted accordingly`,
+        );
+      }
+
+      // If this trade does not conserve rights, it will throw an
+      // error and the user will get their funds back.
+      try {
+        trade(
+          zcf,
+          {
+            seat: marketMakerSeat,
+            gains: price,
+            losses: assets,
+          },
+          { seat, gains: userWantActual, losses: userGiveActual },
+        );
+      } catch (err) {
+        console.log('actual proposal: ', seat.getProposal());
+        console.log('quote: ', {
+          assets,
+          price,
+        });
+        throw Error(
+          'Your offer did not match the quote. See the console log for more details.',
+        );
+      }
+      seat.exit();
+      return 'Trade successful';
+    };
+
+    const customProperties = harden({
+      underlyingAssets: assets,
+      strikePrice: price,
+      byTimerAuthority: timerAuthority,
+      beforeOrAtDeadline: deadline,
+    });
+
+    const acceptQuoteInvitation = zcf.makeInvitation(
+      acceptQuote,
+      'acceptQuote',
+      customProperties,
+    );
+    return acceptQuoteInvitation;
+  };
+
+  const addInventory = seat => {
+    // Take everything in this seat and add it to the marketMakerSeat
+    trade(
+      zcf,
+      { seat: marketMakerSeat, gains: seat.getCurrentAllocation() },
+      { seat, gains: {} },
+    );
+    seat.exit();
+    return 'Inventory added';
+  };
+
+  const removeInventory = seat => {
+    const { want } = seat.getProposal();
+    trade(zcf, { seat: marketMakerSeat, gains: {} }, { seat, gains: want });
+    seat.exit();
+    return 'Inventory removed';
+  };
+
+  const creatorFacet = {
+    // The inventory can be added in bulk before any quotes are made
+    // or can be added immediately before a quote.
+    makeAddInventoryInvitation: async (issuerKeywordRecord = undefined) => {
+      const { issuers } = zcf.getTerms();
+      const issuersPSaved = Object.entries(issuerKeywordRecord).map(
```

Comment:
I think it would be clearer to have specific keywords hardwired. The agnostic version seems harder to understand.  the main contract would just be In/Out, and those could be any two tokens.

Reply:
Looking at your tests, I can see that would not fly. But havign the same API as our AMM might be convenient.